test/java/seedu/addressbook/util/TestUtil.java

```diff
@@ -51,6 +56,48 @@ public static AddressBook createAddressBook(Person... persons) throws DuplicateP
     public static AddressBook clone(AddressBook addressBook) {
         return new AddressBook(addressBook.getAllPersons(), addressBook.getAllTags());
     }
+
+    /**
+     * Returns true iff every pair of corresponding elements two iterables are (deeply) identical.
+     * In other words, the two containers must have the same elements, in the same order.
+     */
+    public static <T> boolean isIdentical(Iterable<T> firstIterable, Iterable<T> secondIterable) {
+        Iterator<T> currentPtr0 = firstIterable.iterator();
+        Iterator<T> currentPtr1 = secondIterable.iterator();
+
+        while (currentPtr0.hasNext() && currentPtr1.hasNext()) {
+            T val0 = currentPtr0.next();
+            T val1 = currentPtr1.next();
+
+            if (!val0.equals(val1)) {
+                return false;
+            }
+        }
+
+        // If any of the two iterables still have elements, then they have different sizes.
+        return !(currentPtr0.hasNext() || currentPtr1.hasNext());
+    }
+    
+    /**
+     * Returns true iff the underlying container behind an iterable is empty.
```

Comment:
Do we need the `iff`? In other places we use `if` even when we mean `if and only if`